packages/just-repo-utils/src/repoInfo.ts

```diff
@@ -25,6 +40,31 @@ export function findGitRoot(cb?: (current: string) => boolean | void): string {
   throw 'No repository root found!';
 }
 
+/**
+ * helper for finding the repository root that also returns the package loader so that it can be
+ * retained for use later.
+ *
+ * @param cb - standard findGitRoot/findRepoRoot callback
+ */
+function findRepoRootWithConfig(cb?: FindRootCallback): [string, PackageJsonLoader | undefined] {
+  let loader: PackageJsonLoader | undefined = undefined;
+  const path = findGitRoot(cwd => {
+    loader = getConfigLoader<PackageJson>(cwd, 'package.json');
+    return (cb && cb(cwd)) || isRepoRoot(cwd, loader && loader());
+  });
+  return [path, loader];
```

Comment:
nit: I personally prefer return objects instead so you know what it is. Array result types are a bit mysterious as a consumer.

Reply:
I definitely agree for public APIs.  It was just fractionally quicker since this is an internal API.  In the future I'll return typed objects.